packages/core/src/artifacts.ts

```diff
@@ -609,9 +609,10 @@ const makeExecutionArtifact = async (
   merkleRoot: string,
   provider: SphinxJsonRpcProvider
 ): Promise<ExecutionArtifact> => {
-  const ethersResponses = await Promise.all(
-    receipts.map((rcpt) => provider.getTransaction(rcpt.hash))
-  )
+  const ethersResponses: (ethers.TransactionResponse | null)[] = []
+  for (const rcpt of receipts) {
+    ethersResponses.push(await provider.getTransaction(rcpt.hash))
+  }
```

Comment:
Using `Promise.all` here causes issues if we have a lot of transactions in a deployment. I noticed this causing issues when testing Mean Finances deployment on Polygon ZKEvm